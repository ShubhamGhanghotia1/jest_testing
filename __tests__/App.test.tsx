import 'react-native';
import React from 'react';
import App from '../App';
import { it } from '@jest/globals';
import renderer from 'react-test-renderer';

describe('App Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains a Text element', () => {
    const tree = renderer.create(<App />).toJSON();
    const textElement = tree?.children?.find(
      (node: any) => node.type === 'Text'
    );
    expect(textElement).toBeTruthy();
  });
});
