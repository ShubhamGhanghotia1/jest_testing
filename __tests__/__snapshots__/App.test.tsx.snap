// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<View>
  <Text>
    App
  </Text>
  <Text>
    shubham ghanghotia
  </Text>
</View>
`;
